src/mame/f256/f256.cpp

```diff
@@ -0,0 +1,2008 @@
+ *    All I/O are mapped to slot 6 ($C000-$DFFF) - there are 4 I/O maps, switched using address $0001.
+ *    Sound Chips: OPL3, PSG, SN74689, CODEC
+ *    Keyboard: mechanical switch in a matrix - controlled by VIA6522 chip.
+ *    Joysticks: 2 Atari-type ports and 2 S/NES ports
+ *    Mouse: over PS/2 - which can also be used for PS/2 Keyboard
+ *    IEC: Commodore Floppy Drive Controller
+ *
+ */
+
+f256_state::f256_state(const machine_config &mconfig, device_type type, const char *tag) :
+    driver_device(mconfig, type, tag)
+    , m_maincpu(*this, MAINCPU_TAG)
+    , m_ram(*this, RAM_TAG)
+    , m_iopage0(*this, IOPAGE0_TAG)
+    , m_iopage1(*this, IOPAGE1_TAG)
+    , m_iopage2(*this, IOPAGE2_TAG)
+    , m_iopage3(*this, IOPAGE3_TAG)
+    , m_rom(*this, ROM_TAG)
+    , m_font(*this, FONT_TAG)
+    , m_screen(*this, SCREEN_TAG)
+    , m_rtc(*this, "rtc")
+    , m_keyboard(*this, "ROW%u", 0)  // this, with the 8 array, requires 8 ROW of INPUTs
+    , m_via6522_0(*this, "via6522_0")
+	, m_via6522_1(*this, "via6522_1")
+
+    , m_sn0(*this, "sn76489_0")
+    , m_sn1(*this, "sn76489_1")
+    , m_opl3(*this, "ymf262")
+    , m_sid0(*this, "sid_0")
+    , m_sid1(*this, "sid_1")
+
+    , m_video(*this, "tiny_vicky")
+    //, m_iec(*this, "iec_bus")
+    //, m_iec_data_out(1)
+    //, m_iec_srq(1)
+    , m_ps2_keyboard(*this, "ps2_kbd")
+    , m_mouse(*this, "ps_mouse")
+    , m_uart(*this, "uart")
+    , m_sdcard(*this, "sdcard")
+	, m_spi_clock_state(false)
+	, m_spi_clock_sysclk(false)
+	, m_spi_clock_cycles(0)
+{
+}
+
+void f256_state::f256k(machine_config &config)
+{
+    W65C02(config, m_maincpu, MASTER_CLOCK/4);
+    m_maincpu->set_addrmap(AS_PROGRAM, &f256_state::program_map);
+
+    RAM(config, m_ram).set_default_size("512k").set_default_value(0x0);
+    RAM(config, m_iopage0).set_default_size("8k").set_default_value(0x0);
+    RAM(config, m_iopage1).set_default_size("8k").set_default_value(0x0);
+    RAM(config, m_iopage2).set_default_size("8k").set_default_value(0x0);
+    RAM(config, m_iopage3).set_default_size("8k").set_default_value(0x0);
+
+    SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+    m_screen->set_refresh_hz(60); // Refresh rate (e.g., 60Hz)
+    m_screen->set_size(800,525);
+    //m_screen->set_visarea(160, 799, 45, 524);  // this is how it should reall work, but the screen ends up offset
+    m_screen->set_visarea(0, 639, 0, 479);
+    m_screen->set_screen_update(m_video, FUNC(tiny_vicky_video_device::screen_update));
+    
+    BQ4802(config, m_rtc, MASTER_CLOCK / 1000);  // RTC clock in kHz
+    //set interrupt handler for the RTC
+    m_rtc->int_handler().set(FUNC(f256_state::rtc_interrupt_handler));
+    
+    TINY_VICKY(config, m_video, MASTER_CLOCK);
+    m_video->sof_irq_handler().set(FUNC(f256_state::sof_interrtupt));
+    m_video->sol_irq_handler().set(FUNC(f256_state::sol_interrtupt));
+
+    MOS6522(config, m_via6522_0, MASTER_CLOCK / 4);  // Atari Joysticks
```

Comment:
I believe this should be using the W65C22S variant (though MAME does emulate the subtle differences between VIA variants at present).

Reply:
I could not confirm this, but it is a surface mount chip, so a recent VIA chip.  So I've made the change.